src/overlap_check/test/overlap_check_test.cpp

```diff
@@ -1,17 +1,15 @@
-#include <iostream>
+#include "overlap_check_test.hpp"
+
+#include <algorithm>
 #include <cmath>
 #include <cstdlib>
 #include <ctime>
-#include <vector>
+#include <iostream>
 #include <set>
-#include <algorithm>
-#include <memory>
-#include "moab/Core.hpp"
+#include <vector>
 
 #include "overlap.hpp"
 
-#include "overlap_check_test.hpp"
-
 void OverlapTest::SetUp() {
```

Comment:
reordering header broke compilation...

`memory` and `moab/Core` were included before `overlap_check_test.hpp` so compilation was ok, events they were needed by `overlap_check_test.hpp` and not included there...
I moved them in `overlap_check_test.hpp`....